java/rocksjni/parsed_entry_info.cc

```diff
@@ -0,0 +1,216 @@
+                                                         jlong handle,
+                                                         jlong options_handle,
+                                                         jbyteArray jtarget,
+                                                         jint len) {
+  auto *options =
+      reinterpret_cast<const ROCKSDB_NAMESPACE::Options*>(options_handle);
+  auto *parsed_entry_info = reinterpret_cast<ROCKSDB_NAMESPACE::ParsedEntryInfo*>(handle);
+  jbyte* target = env->GetByteArrayElements(jtarget, nullptr);
+  if (target == nullptr) {
+  // exception thrown: OutOfMemoryError
+    return;
+  }
+  ROCKSDB_NAMESPACE::Slice target_slice(reinterpret_cast<char*>(target), len);
+  ROCKSDB_NAMESPACE::ParseEntry(target_slice, options->comparator,
+                                parsed_entry_info);
+}
+
+/*
+ * Class:     org_rocksdb_ParsedEntryInfo
+ * Method:    parseEntryDirect
+ * Signature: (JJLjava/nio/ByteBuffer;II)V
+ */
+void JNICALL Java_org_rocksdb_ParsedEntryInfo_parseEntryDirect(JNIEnv *env,
+                                                               jclass clz,
+                                                               jlong handle,
+                                                               jlong options_handle,
+                                                               jobject jbuffer,
+                                                               jint jbuffer_off,
+                                                               jint jbuffer_len) {
+  auto *options =
+      reinterpret_cast<const ROCKSDB_NAMESPACE::Options*>(options_handle);
+  auto *parsed_entry_info =
+      reinterpret_cast<ROCKSDB_NAMESPACE::ParsedEntryInfo*>(handle);
+  auto parse = [&parsed_entry_info, &options](
+                   ROCKSDB_NAMESPACE::Slice& target_slice) {
+    ROCKSDB_NAMESPACE::ParseEntry(target_slice, options->comparator,
+                                  parsed_entry_info);
+  };
+  ROCKSDB_NAMESPACE::JniUtil::k_op_direct(parse, env, jbuffer, jbuffer_off,
+                                          jbuffer_len);
+}
+
+/*
+ * Class:     org_rocksdb_ParsedEntryInfo
+ * Method:    parseEntryByteArray
+ * Signature: (JJ[BII)V
+ */
+void JNICALL Java_org_rocksdb_ParsedEntryInfo_parseEntryByteArray(JNIEnv *env,
+                                                                  jclass clz,
+                                                                  jlong handle,
+                                                                  jlong options_handle,
+                                                                  jbyteArray jtarget,
+                                                                  jint off,
+                                                                  jint len) {
+  const std::unique_ptr<char[]> target(new char[len]);
+  if (target == nullptr) {
+    jclass oom_class = env->FindClass("/lang/java/OutOfMemoryError");
```

Comment:
I don't think this has been tested as the class name is incorrect. Also there are util methods in portal.h for throwing errors that could be used.

Reply:
Added util method to also do operations on indirect byte array buffers